mm2src/coins/hd_wallet/coin_ops.rs

```diff
@@ -189,7 +182,7 @@ pub trait HDWalletCoinOps {
 
         let trezor_coin = self.trezor_coin()?;
         let derivation_path = hd_address.derivation_path().clone();
-        let expected_address = hd_address.address().to_string();
+        let expected_address = hd_address.address().display_address();
```

Comment:
@dimxy according to [this](https://github.com/trezor/connect/blob/develop/docs/methods/ethereumGetAddress.md#migration-from-older-version), using to_string was a bug if we want to support v5

Reply:
Oh, there are more code where this needs to be fixed for trezor:
[here](https://github.com/KomodoPlatform/komodo-defi-framework/blob/365858790ecba14076077cd579f911721835c2be/mm2src/trezor/src/eth/eth_command.rs#L254)
and [here](https://github.com/KomodoPlatform/komodo-defi-framework/blob/365858790ecba14076077cd579f911721835c2be/mm2src/trezor/src/eth/eth_command.rs#L204)

(shouldn't we use display_address everywhere in code for simplicity?)

Reply:
The refactor of HD wallet code to a crate is outside of this PR scope so I will add it to this checklist https://github.com/KomodoPlatform/komodo-defi-framework/issues/1838#issuecomment-1675467570

Reply:
btw we have the mm2_eth crate. It's used only for metamask now.
But maybe we can move all eth types/primitives related stuff there, so other crates including hd wallet may get them from it, like it is now for the btc primitives

Reply:
> DisplayAddress is related to wallet operations and it makes sense for it to be part of a wallet crate (which HD wallet will be).

`AddrToString` trait should be moved to wallet crate as well then, and leave it close to `DisplayAddress` 

https://github.com/KomodoPlatform/komodo-defi-framework/blob/dce9da850183ef2d19131f7eac399a20c6b32725/mm2src/coins/lp_coins.rs#L1602-L1610